src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -323,6 +339,87 @@ private static IEnumerable<VisualStudioInstance> GetInstances(VisualStudioInstan
 #endif
         }
 
+        static IEnumerable<VisualStudioInstance> GetMonoMSBuildInstances ()
+        {
+            // $prefix/lib/mono/4.5/mscorlib.dll
+            var runningMonoPath = Path.GetFullPath(Path.Combine(Path.GetDirectoryName(typeof (object).Assembly.Location), "..", "..", ".."));
+            if (TryGetValidMonoVersion (runningMonoPath, out var version))
+            {
+                yield return new VisualStudioInstance("Mono", runningMonoPath, version, DiscoveryType.Mono);
+            }
+
+            if (!IsOSX)
+            {
+                // Returning just one instance on !osx
+                yield break;
+            }
+
+            foreach(var dirPath in Directory.EnumerateDirectories(s_monoOSXBasePath))
```

Comment:
do we really needs this? `mono --version=number` should work fine on OSX.

Reply:
Switched to using `mono --version=number`. On OSX, doing this for all the versions found.